gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -699,14 +699,18 @@ private final PreparedStatement prepMinion(
 						qt.bindStatement();
 					}
 					catch(StandardException | AssertFailure ex) {
-						if (routeQuery && !DML_TABLE_PATTERN.matcher(source).matches()) {
-                                                       if (observer != null) {
-                                                         observer.testExecutionEngineDecision(qinfo, ExecutionEngine.SPARK, this.statementText);
-                                                       }
+					  if (routeQuery) {
+					    if (observer != null) {
+					      observer.testExecutionEngineDecision(qinfo, ExecutionEngine.SPARK, this.statementText);
+					    }
+					    boolean isUpdateOrDelete = false;
+					    if (DML_TABLE_PATTERN.matcher(source).matches()) {
+					    	isUpdateOrDelete = true;
+					    }
 							return getPreparedStatementForSnappy(true, statementContext, lcc, false,
-                  checkCancellation, false);
-						}
-						throw ex;
+							  checkCancellation, isUpdateOrDelete);
+					  }
+					  throw ex;
```

Comment:
indentation looks bit out-of-sync with previous line

Reply:
This file is too bad for indentation rules :-) ... Corrected few things but this remained. May be will take care of this later.